contracts/timealarms/src/alarms.rs

```diff
@@ -66,18 +69,24 @@ impl TimeAlarms {
         self.time_alarms
             .alarms_selection(storage, ctime)
             .take(max_count.try_into()?)
+            .collect::<Result<Vec<_>, _>>()?
+            .into_iter()
```

Comment:
Needed in order to be able to use the `dyn Storage` after loading them, to remove the alarm.
First borrow occurs when loading and second one, which is mutable, occurs during the folding transformation.

Reply:
I see ... the cause comes from the lack of 'modify-as-we-go' functionality of cosmwasm iterators.

The following will not fix the issue but will simplify the code:
- move the `storage` as a member variable of the `time_oracle::Alarms`
- merge InDelivery into `time_oracle::Alarms` thus avoiding passing the time back and forth. Instead of calling `IndexedMap::remove` use `IndexedMap::may_load`, `IndexedMap::replace` and `Deque::push_back`

Reply:
That's not the problem, the real problem is that the `Alarms::alarms_selection` method borrows the storage. The position of the `storage` doesn't matter in this exact case.

Reply:
right, and my comment does not claim that would fix it